### PR TITLE
Refactor solvers.jl

### PR DESCRIPTION
**Pull request by joehuchette**

### CODE REVIEW

**Comment by joehuchette** [comment]
I'd like to merge this soon so that I can rebase the norm branch on top of it.

**Comment by mlubin** [comment]
Will take a look later tonight

File: src/solvers.jl [hunk 1]
```diff
@@ -31,31 +39,69 @@ function solve(m::Model; suppress_warnings=false, ignore_solve_hook=(m.solvehook
...
+    if !(stat == :Infeasible || stat == :Unbounded)
+        try
+            objVal = MathProgBase.getobjval(m.internalModel) + m.obj.aff.constant
+            colVal = MathProgBase.getsolution(m.internalModel)
+            m.objVal = objVal # Don't corrupt the answers if one of the above two calls fails
+            m.colVal = colVal
+        end
+    end
+
+    if traits.sdp && m.objSense == :Max
```
Comment (mlubin) [inline src/solvers.jl]: Not ideal that the logic for the objective flip leaks through here, but not a big deal

Comment (joehuchette) [inline src/solvers.jl]: Agreed, but there's not really a way around that without requiring conic solvers to implement ``setsense!`` (which I still think is a good idea).

**Comment by mlubin** [comment]
Looks good to merge

**Pull request merged**